api/src/api/diag.ts

```diff
@@ -67,7 +71,8 @@ export class DiagAPI implements DiagLogger {
 
     self.setLogger = (
       logger: DiagLogger,
-      logLevel: DiagLogLevel = DiagLogLevel.INFO
+      logLevel: DiagLogLevel = DiagLogLevel.INFO,
+      options: LoggerOptions = {}
```

Comment:
You also need to update the types https://github.com/open-telemetry/opentelemetry-js/pull/3366/files#diff-27d843b1252efc09aad875a53b4b2bb9a4b98180667b1669d46067af08662bedR125

Reply:
Wonder if we want to add a third parameter or make the second parameter an overload?

Reply:
updated the type

Reply:
happy to make it an overload if you want? 🙂

Reply:
We are avoiding overloading API methods that need to be implemented in SDK libraries. However, this method is not part of an interface that needs to be implemented in SDKs.

So I believe it would be intuitive to be overloaded as an option bag.

Reply:
I attempted adding that now, but I'm getting some type errors. I'm not sure why the methods are assigned in the constructor instead of on the class directly. So I pushed a second commit that moves it and it resolves the type error.

Help appreciated 😅

Reply:
If moving away from `self.setLogger` to defining "normally" on the class is acceptable, I'd like to land that in a separate PR to reduce the noise of this one 🙂 If it's _not_ acceptable, I'll need some help.

Reply:
the self.logger use is done there intentionally to aid with code minification

Reply:
if you revert the last commit, you can see the typescript errors

Reply:
> if you revert the last commit, you can see the typescript errors

The compilation errors can be suppressed if the two method declarations are merged into a single one, e.g.

```ts
  /**
   * Set the global DiagLogger and DiagLogLevel.
   * If a global diag logger is already set, this will override it.
   *
   * @param logger - [Optional] The DiagLogger instance to set as the default logger.
   * @param logLevelOrOptions - [Optional] And object which can contain `logLevel: DiagLogLevel` used to filter logs sent to the logger. If not provided it will default to INFO. The object may also contain `suppressOverrideMessage: boolean` which will suppress the warning normally logged when a logger is already registered.
   * @returns true if the logger was successfully registered, else false
   */
  public setLogger!: (logger: DiagLogger, logLevelOrOptions?: DiagLogLevel | LoggerOptions) => boolean;
```

As for why those methods need to be defined in this non-conventional way, I traced back to https://github.com/open-telemetry/opentelemetry-js/pull/1880 and didn't find an explanation. Maybe @MSNev can chime in and share more about it?

Reply:
Ok, I've blocked this PR as we should NOT convert the property into a function as this becomes less compressible through minification.

And then will cause me to re-implement this original change again in the sandbox to have it contributed back here.

Reply:
This pattern has major implications on the overall browser bundle sizes (depending on the module format and ES level being targetted). And this is one of the techniques that I WILL be investigating the viability and performing across the entire JS and Contrib codebases in the sandbox (when I get it going -- HOPEFULLY shortly)